### PR TITLE
Prevent unauthorized players from changing the noteblock sound.

### PR DESCRIPTION
**Pull request by auouymous**

### CODE REVIEW

File: mesecons_noteblock/init.lua [hunk 1]
```diff
@@ -3,7 +3,11 @@ minetest.register_node("mesecons_noteblock:noteblock", {
 	tiles = {"mesecons_noteblock.png"},
 	is_ground_content = false,
 	groups = {snappy=2, choppy=2, oddly_breakable_by_hand=2},
-	on_punch = function(pos, node) -- change sound when punched
+	on_punch = function(pos, node, puncher) -- change sound when punched
+		if minetest.is_protected(pos, puncher and puncher:get_player_name()) then
+			return
```
Comment (Panquesito7) [inline mesecons_noteblock/init.lua:8]:
```suggestion
			minetest.record_protection_violation(pos, puncher)
			return
```

Comment (auouymous) [reply mesecons_noteblock/init.lua:8]: The delayer doesn't record it, and should accidentally punching a block record a violation?

Comment (SmallJoker) [reply mesecons_noteblock/init.lua:8]: No. The PR is good as-is.

**Review by SmallJoker — APPROVED** [review]
Will merge in a few days unless someone else does that.

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**